pkg/providers/nutanix/config/cp-template.yaml

```diff
@@ -0,0 +1,196 @@
+piVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: NutanixCluster
+metadata:
+  name: "{{.clusterName}}"
+  namespace: "{{.eksaSystemNamespace}}"
+spec:
+  prismCentral:
+    address: "{{.nutanixEndpoint}}"
+    port: {{.nutanixPort}}
+    insecure: true
+    credentialRef:
+      name: "{{.clusterName}}"
+      kind: Secret
+  controlPlaneEndpoint:
+    host: "{{.controlPlaneEndpointIp}}"
+    port: 6443
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: "{{.clusterName}}"
+  namespace: "{{.eksaSystemNamespace}}"
+stringData:
+  NUTANIX_PASSWORD: "{{.nutanixPassword}}"
+  NUTANIX_USER: "{{.nutanixUser}}"
+---
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  labels:
+    cluster.x-k8s.io/cluster-name: "{{.clusterName}}"
+  name: "{{.clusterName}}"
+  namespace: "{{.eksaSystemNamespace}}"
+spec:
+  clusterNetwork:
+    services:
+      cidrBlocks: {{.serviceCidrs}}
+    pods:
+      cidrBlocks: {{.podCidrs}}
+    serviceDomain: "cluster.local"
+  controlPlaneRef:
+    apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+    kind: KubeadmControlPlane
+    name: "{{.clusterName}}"
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+    kind: NutanixCluster
+    name: "{{.clusterName}}"
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+kind: KubeadmControlPlane
+metadata:
+  name: "{{.clusterName}}"
+  namespace: "{{.eksaSystemNamespace}}"
+spec:
+  replicas: {{.controlPlaneReplicas}}
+  version: "{{.kubernetesVersion }}"
+  machineTemplate:
+    infrastructureRef:
+      apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+      kind: NutanixMachineTemplate
+      name: "{{.clusterName}}-mt-0"
+  kubeadmConfigSpec:
+    clusterConfiguration:
+      imageRepository: "{{.kubernetesRepository}}"
+      apiServer:
+        certSANs:
+          - localhost
+          - 127.0.0.1
+          - 0.0.0.0
+      controllerManager:
+        extraArgs:
+          enable-hostpath-provisioner: "true"
+      dns:
+        imageRepository: {{.corednsRepository}}
+        imageTag: {{.corednsVersion}}
+      etcd:
+  {{- if .externalEtcd }}
+external:
+  endpoints: []
+  caFile: "/etc/kubernetes/pki/etcd/ca.crt"
+  certFile: "/etc/kubernetes/pki/apiserver-etcd-client.crt"
+  keyFile: "/etc/kubernetes/pki/apiserver-etcd-client.key"
+  {{- else }}
+local:
+  imageRepository: {{.etcdRepository}}
+  imageTag: {{.etcdImageTag}}
+  {{- end }}
+files:
```

Comment:
The indentation looks off here

Reply:
Good catch, the `if/else` blocks seem to be messing with my editor.